Tests/KituraTests/KituraTest.swift

```diff
@@ -63,41 +92,61 @@ class KituraTest: XCTestCase {
     }
 
-    func performServerTest(_ router: ServerDelegate, sslOption: SSLOption = SSLOption.both, timeout: TimeInterval = 10,
+    func performServerTest(_ router: ServerDelegate, sslOption: SSLOption = SSLOption.both, socketTypeOption: SocketTypeOption = SocketTypeOption.both, timeout: TimeInterval = 10,
                            line: Int = #line, asyncTasks: (XCTestExpectation) -> Void...) {
-        performServerTest(router, sslOption: sslOption, timeout: timeout, line: line, asyncTasks: asyncTasks)
+        performServerTest(router, sslOption: sslOption, socketTypeOption: socketTypeOption, timeout: timeout, line: line, asyncTasks: asyncTasks)
     }
 
-    func performServerTest(_ router: ServerDelegate, sslOption: SSLOption = SSLOption.both, timeout: TimeInterval = 10,
+    func performServerTest(_ router: ServerDelegate, sslOption: SSLOption = SSLOption.both, socketTypeOption: SocketTypeOption = SocketTypeOption.both, timeout: TimeInterval = 10,
                            line: Int = #line, asyncTasks: [(XCTestExpectation) -> Void]) {
         if sslOption != SSLOption.httpsOnly {
-            self.port = KituraTest.httpPort
             self.useSSL = false
-            doPerformServerTest(router: router, timeout: timeout, line: line, asyncTasks: asyncTasks)
+            if socketTypeOption != SocketTypeOption.unix {
+                self.useUnixSocket = false
+                doPerformServerTest(router: router, timeout: timeout, line: line, asyncTasks: asyncTasks)
+            }
+            setUp()
+            if socketTypeOption != SocketTypeOption.inet {
+                self.useUnixSocket = true
+                doPerformServerTest(router: router, timeout: timeout, line: line, asyncTasks: asyncTasks)
+            }
         }
         
         // Call setUp to start at a known state (ideally, this should have been written as a separate test)
         setUp()
 
         if sslOption != SSLOption.httpOnly {
-            self.port = KituraTest.httpsPort
             self.useSSL = true
-            doPerformServerTest(router: router, timeout: timeout, line: line, asyncTasks: asyncTasks)
+            if socketTypeOption != SocketTypeOption.unix {
+                self.useUnixSocket = false
+                doPerformServerTest(router: router, timeout: timeout, line: line, asyncTasks: asyncTasks)
+            }
+            setUp()
+            if socketTypeOption != SocketTypeOption.inet {
+                self.useUnixSocket = true
+                doPerformServerTest(router: router, timeout: timeout, line: line, asyncTasks: asyncTasks)
+            }
         }
     }
 
     func doPerformServerTest(router: ServerDelegate, timeout: TimeInterval, line: Int, asyncTasks: [(XCTestExpectation) -> Void]) {
 
-        guard let port = startServer(router: router) else {
-            XCTFail("Error starting server. useSSL:\(useSSL)")
-            return
+        if useUnixSocket {
```

Comment:
`self.`? And throughout the PR.. I think it's a good practice.

Reply:
I changed this to use a private enum: https://github.com/IBM-Swift/Kitura/pull/1434/commits/026f967f56e4300f21cf5e740dd59c3a5d1e4615